rosbag2_py/src/rosbag2_py/_transport.cpp

```diff
@@ -204,6 +251,38 @@ class Recorder
       SIGINT, [](int /* signal */) {
         rosbag2_py::Recorder::cancel();
       });
+#ifdef _WIN32
```

Comment:
If we were to repeat this block a third time - I'd suggest that we pull it out somehow but for now in just 2 locations, it's OK

Reply:
Yeah, I had the same thoughts. But decided to keep it in place. It is just 1 line inside and with lambda, it looks more clear what is going on inside the signal handler.